Taylor2

iPopcorn opened the pull request:
I have added 2 functions, a few global variables, and a struct. These functions convert a path array to a list of commands, and then executes the commands.

iPopcorn committed:
Added Function Path_To_Robot

Function Name: Path_To_Robot
Description: Given an array of points, convert it to an array of
instructions.
Parameters: path_length is the length of the g_path array.
Returns: int[] instruction_list

iPopcorn committed:
Execute_Path

Execute_Path reads the instruction array and executes the path from
start to finish. Just have to replace the print statements with robot
control functions.

jinckse commented:
We may have to rethink some things after yesterday's class since we now know that everything absolutely must be run on the robot. As far as path planning goes. I was thinking about the possibility of us just doing the path planning on the fly, and just at least keeping the Manhattan mapping function. Other than that, we'll have to look into smaller data types, and getting rid of all unneeded variables and arrays (too much memory).

I think the best thing at this point is to just focus on the specific movement instructions, independently of following a certain path right now, then those movements can be linked to the path decisions as it goes.

iPopcorn commented:
The logic for the functions can easily be adapted to a path generated on the fly. We just have to keep track of the robot's current direction. Once we check the 4 adjacent points, the robot can can decide what moves it needs to make based on its current direction and the location of the point. For example, if the next point in the path is +1 in the x direction, and the robot is facing north, then the robot will know that it needs to turn right, then move, in order to reach the next point. So the logic is the same, check if the next point is north, south, east, or west of where the robot is, and then move based on the robot's current direction.

iPopcorn committed:
Manhattan Not Working

Here is the manhattan algorithm with the course we were testing in the
lab.  It looks like its wrapping over the side to find the path.

GregoryStout committed:
Create Last_Chance
